replacer.go

```diff
@@ -354,7 +354,15 @@ func (f fileReplacementProvider) replace(key string) (any, bool) {
 			zap.Error(err))
 		return nil, true
 	}
-	return string(body), true
+	content := string(body)
+	// Remove a single trailing newline, regardless of its type
+	if strings.HasSuffix(content, "\r\n") {
+		return strings.TrimSuffix(content, "\r\n"), true
```

Comment:
Why do we call `HasSuffix`? `TrimSuffix` is only effective if the input has that suffix.

Reply:
This way, we make sure to remove only the very last newline due to `HasSuffix` in combination with `return strings.TrimSuffix`.

For example, let's have this content in a file:
```
hello world\r\n
\r\n
\r\n
```

What we want:
```
hello world\r\n
\r\n
```

And this code without `HasSuffix` :
```go
// Remove a single trailing newline, regardless of its type
content = strings.TrimSuffix(content, "\r\n")
content = strings.TrimSuffix(content, "\n")
return strings.TrimSuffix(content, "\r"), true
```

content after 1. TrimSuffix:
```
hello world\r\n
\r\n
```

content after 2. TrimSuffix:
```
hello world\r\n
\r
```

return after 3. TrimSuffix:
```
hello world\r\n
```

Reply:
Can't you just do 
```go
content = strings.TrimSuffix(content, "\n")
content = strings.TrimSuffix(content, "\r")
return content
```
?

Also technically for windows the convention is different from unix platforms, so removing the newline there is kinda redundant, but like all good software developers we can just ignore that fact.

Reply:
> When these two test files are checked out on the Windows platform for the Windows test cases

But the test cases are to reflect users' files as-is on Windows, which may have `\r\n`. I think configuring that in gitignore is just cheating the test, no?

Reply:
But on windows, unlike on Linux due to the unix convention (see also https://superuser.com/questions/745111/why-is-vim-adding-a-newline-is-this-a-convention/745135#745135, you don't tpyically have an trailing newline for a file.

Reply:
I think the concern is that if you author the files on your local Windows machine, then you upload to a Linux server, the Windows-style line endings are relevant because they don't just get converted automatically.

Reply:
If you author the code in windows machine using a software that follows windows conventions and don't just add a training newline yourself then you will get a file without \r\n at the end.

The real concern is when that is not the case, for example someone shares you a file with unix conventions, and it gets converted to windows ones because your editor is bad, or you just do something stupid with the tooling for that to happen. 

Anyway I think it is correct to remove line ending there because: 1. current error for mismatched acme key is not that much helpful 2. if you use windows you should expect such bullshit to happen, just cope with it 3. usecase of having working secrets outweighs the usecase of displaying a file with a single empty line at the end. 4. it requires less thinking

Reply:
I concede to kanashimia's point 👍🏼